### PR TITLE
Add Misssing Type Annotations

### LINKED ISSUE #15: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_max31856.py:149
- [ ] adafruit_max31856.py:207
- [ ] adafruit_max31856.py:229
- [ ] adafruit_max31856.py:282
- [ ] adafruit_max31856.py:291

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #15
submitted for review / comment

### CODE REVIEW

File: adafruit_max31856.py [hunk 1]
```diff
@@ -208,7 +217,7 @@ def noise_rejection(self) -> int:
         return 60
 
     @noise_rejection.setter
-    def noise_rejection(self, frequency: int):
+    def noise_rejection(self, frequency: int) -> None:
```
Comment (tekktrik) [inline adafruit_max31856.py]: You actually annotated the return correctly, but while we're editing this file, let's make the getter and setter for `noise_rejection` not be `int` but rather `Literal[50, 60]`!

File: adafruit_max31856.py [hunk 2]
```diff
@@ -279,7 +288,9 @@ def temperature_thresholds(self, val):
         self._write_u8(_MAX31856_LTLFTL_REG, int_low)
 
     @property
-    def reference_temperature_thresholds(self):  # pylint: disable=invalid-name
+    def reference_temperature_thresholds(  # pylint: disable=invalid-name,
+        self,
+    ) -> Tuple[float, float]:  # pylint: disable=invalid-name
```
Comment (tekktrik) [inline adafruit_max31856.py]: Multiline `pylint` is a pain.  You can get rid of on of these `# pylint: disable=invalid-name` comments, and I think it's THIS one.

File: adafruit_max31856.py [hunk 3]
```diff
@@ -289,13 +300,15 @@ def reference_temperature_thresholds(self):  # pylint: disable=invalid-name
         )
 
     @reference_temperature_thresholds.setter
-    def reference_temperature_thresholds(self, val):  # pylint: disable=invalid-name
+    def reference_temperature_thresholds(  # pylint: disable=invalid-name,
+        self, val: Tuple[float, float]
+    ) -> None:  # pylint: disable=invalid-name
```
Comment (tekktrik) [inline adafruit_max31856.py]: Same thing here, I think it's this second `# pylint: disable=invalid-name` you can get rid of.

File: adafruit_max31856.py [hunk 4]
```diff
@@ -358,11 +371,11 @@ def oneshot_pending(self) -> bool:
         )
         return bool(oneshot_flag)
 
-    def _wait_for_oneshot(self):
+    def _wait_for_oneshot(self) -> None:
         while self.oneshot_pending:
             sleep(0.01)
 
-    def _read_register(self, address, length):
+    def _read_register(self, address: int, length: int) -> int:
```
Comment (tekktrik) [inline adafruit_max31856.py]: Since it's actually returning a slice of a `bytearray`, the return here should be `bytearray`.

**Review by tekktrik — CHANGES_REQUESTED** [review]
A few bits of feedback, thanks for the PR!

File: adafruit_max31856.py (hunk 1 above)
Comment (tcfranks) [reply adafruit_max31856.py]: rats! missed that in the docstring!

File: adafruit_max31856.py (hunk 2 above)
Comment (tcfranks) [reply adafruit_max31856.py]: I think originally they were correct and black broke up the line after the annotations were added....I missed the dup when I added the corrections....

**Review by tekktrik — APPROVED** [review]
Awesome, thanks for another great PR!

File: adafruit_max31856.py (hunk 2 above)
Comment (tekktrik) [reply adafruit_max31856.py]: Yeah, black breaks moves it to the bottom line, but then pylint just gets confused again.  It's one of the few bad pairings of the two I commonly run into myself.

**Pull request merged**